google/cloud/aiplatform/training_jobs.py

```diff
@@ -66,6 +66,290 @@
+            container_uri (str):
+                Required: Uri of the training container image in the GCR.
+            project (str):
+                Optional project to retrieve model from. If not set, project set in
+                aiplatform.init will be used.
+            location (str):
+                Optional location to retrieve model from. If not set, location set in
+                aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional credentials to use to retrieve the model.
+        """
+
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._display_name = display_name
+        self._container_uri = container_uri
+        self._project = project
+        self._credentials = credentials
+        self._gca_resource = None
+
+    def run_job(
+        self,
+        training_task_definition: str,
+        training_task_inputs: dict,
+        dataset: Optional[datasets.Dataset],
+        training_fraction_split: float,
+        validation_fraction_split: float,
+        test_fraction_split: float,
+        model: Optional[gca_model.Model] = None,
+        base_output_dir: Optional[str] = None,
+    ):
+        """Runs the training job.
+
+        Args:
+            training_task_definition (str):
+                Required. A Google Cloud Storage path to the
+                YAML file that defines the training task which
+                is responsible for producing the model artifact,
+                and may also include additional auxiliary work.
+                The definition files that can be used here are
+                found in gs://google-cloud-
+                aiplatform/schema/trainingjob/definition/. Note:
+                The URI given on output will be immutable and
+                probably different, including the URI scheme,
+                than the one given on input. The output URI will
+                point to a location where the user only has a
+                read access.
+            training_task_inputs (~.struct.Value):
+                Required. The training task's parameter(s), as specified in
+                the
+                ``training_task_definition``'s
+                ``inputs``.
+            dataset (aiplatform.Dataset):
+                Optional. The dataset within the same Project from which data will be used to train the Model. The
+                Dataset must use schema compatible with Model being trained,
+                and what is compatible should be described in the used
+                TrainingPipeline's [training_task_definition]
+                [google.cloud.aiplatform.v1beta1.TrainingPipeline.training_task_definition].
+                For tabular Datasets, all their data is exported to
+                training, to pick and choose from.
+            training_fraction_split (float):
+                The fraction of the input data that is to be
```

Comment:
@sasha-gitg sometimes I see `Required` and sometimes `Optional` and sometimes nothing is written here. Any guidance on why nothing is written for this parameter (I copied it from CustomTrainingJobClass).

Reply:
These are generally copied from the protos if they represent the same field: https://github.com/googleapis/python-aiplatform/blob/dev/google/cloud/aiplatform_v1beta1/types/training_pipeline.py#L260

So we inherit the arg commenting from the service.

For comments we add, it's only necessary we mark them as Required if they are indeed so.